### PR TITLE
[INLONG-8396][Manager] Support for querying audit data with average delay

### LINKED ISSUE #8396: [Improve][Manager] Support for querying audit data with average delay
### Description

Support for querying audit data with average delay.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #8396 

### Motivation

1.Support for querying audit data with average delay.
2.Support querying audit data for multiple days.

### Modifications

1.Support for querying audit data with average delay.
2.Support querying audit data for multiple days

### CODE REVIEW

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditInfo.java [hunk 1]
```diff
@@ -30,12 +30,15 @@ public class AuditInfo {
...
     @ApiModelProperty(value = "Audit count")
     private Long count;
+    @ApiModelProperty(value = "Audit delay")
+    private Long delay;
 
     public AuditInfo() {
     }
 
-    public AuditInfo(String logTs, Long count) {
+    public AuditInfo(String logTs, Long count, Long delay) {
```
Comment (vernedeng) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditInfo.java:39]: what if count or delay is null?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 2]
```diff
@@ -268,6 +271,7 @@ private List<String> getAuditIds(String groupId, String streamId, String sinkNod
         if (sinkNodeType == null) {
             auditSet.add(getAuditId(ClusterType.DATAPROXY, true));
         } else {
+            auditSet.add(getAuditId(sinkNodeType, true));
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:274]: Audit supports not only send or receive, please make this interface more common

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java]: done.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditInfo.java (hunk 1 above)
Comment (fuweng11) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditInfo.java:39]: Default 0.

**Review by gong — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 3]
```diff
@@ -205,20 +205,21 @@ public List<AuditVO> listByCondition(AuditRequest request) throws Exception {
                 String format = "%Y-%m-%d %H:%i:00";
                 // Support min agg at now
                 DateTimeFormatter forPattern = DateTimeFormat.forPattern("yyyy-MM-dd");
-                DateTime dtDate = forPattern.parseDateTime(request.getDt());
-                String eDate = dtDate.plusDays(1).toString(forPattern);
+                DateTime edtDate = forPattern.parseDateTime(request.getEndDate());
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:208]: What is `edt` here?

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:208]: Fix.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 4]
```diff
@@ -205,20 +205,21 @@ public List<AuditVO> listByCondition(AuditRequest request) throws Exception {
                 String format = "%Y-%m-%d %H:%i:00";
                 // Support min agg at now
                 DateTimeFormatter forPattern = DateTimeFormat.forPattern("yyyy-MM-dd");
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:207]: `DateTimeFormatter` is thread-safe, please extract it as a static and final constant in this class, thanks.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 5]
```diff
@@ -205,20 +205,21 @@ public List<AuditVO> listByCondition(AuditRequest request) throws Exception {
...
-                DateTime dtDate = forPattern.parseDateTime(request.getDt());
-                String eDate = dtDate.plusDays(1).toString(forPattern);
+                DateTime edtDate = forPattern.parseDateTime(request.getEndDate());
+                String endDate = edtDate.plusDays(1).toString(forPattern);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:209]: May change to `DateTime endDate` and `String endDateStr` is better.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditRequest.java [hunk 6]
```diff
@@ -50,9 +50,11 @@ public class AuditRequest {
     @ApiModelProperty(value = "sink id")
     private Integer sinkId;
 
-    @ApiModelProperty(value = "query date, format by 'yyyy-MM-dd'", required = true, example = "2022-01-01")
-    @NotBlank(message = "dt not be blank")
-    private String dt;
+    @ApiModelProperty(value = "query start date, format by 'yyyy-MM-dd'", required = true, example = "2022-01-01")
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditRequest.java:53]: Those two params are required? Can we be compatible, and only query the data of the last 1 day, 2 days or 7 days if it is not provided?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 7]
```diff
@@ -373,6 +376,7 @@ private List<AuditVO> doAggregate(List<AuditVO> auditVOList, String format) {
         for (AuditVO auditVO : auditVOList) {
             AuditVO statInfo = new AuditVO();
             ConcurrentHashMap<String, AtomicLong> countMap = new ConcurrentHashMap<>();
+            ConcurrentHashMap<String, AtomicLong> delayMap = new ConcurrentHashMap<>();
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:379]: Local variables inside the method do not need to use ConcurrentHashMap, just change it to HashMap.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditRequest.java (hunk 6 above)
Comment (fuweng11) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/audit/AuditRequest.java:53]: Those two params are required.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java (hunk 7 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java (hunk 5 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java (hunk 4 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java]: Done.

**Review by healchow — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**